synthesizer/src/process/execute.rs

```diff
@@ -208,15 +204,27 @@ impl<N: Network> Process<N> {
 
             // Retrieve the verifying key.
             let verifying_key = self.get_verifying_key(stack.program_id(), function.name())?;
-            // Ensure the transition proof is valid.
-            ensure!(
-                verifying_key.verify(&function.name().to_string(), &inputs, transition.proof()),
-                "Transition is invalid - failed to verify transition proof"
-            );
-
-            lap!(timer, "Verify transition proof for {}", function.name());
+            // Save the verifying key and its inputs.
+            verifier_inputs
+                .entry(Locator::new(*stack.program_id(), *function.name()))
+                .or_insert((verifying_key, vec![]))
+                .1
+                .push(inputs);
+
+            lap!(timer, "Constructed the verifier inputs for a transition of {}", function.name());
         }
 
+        // Count the number of verifier instances.
+        let num_instances = verifier_inputs.values().flat_map(|(_, inputs)| inputs).count();
```

Comment:
```suggestion
        let num_instances = verifier_inputs.values().map(|(_, inputs)| inputs.len()).sum::<usize>();
```
this should be faster, as `count` has to consume the whole iterator one-by-one, while the lengths are immediately available